File: 291/a.java
import java.util.Scanner;
import java.util.Arrays;

public class a {

    public static void main(String[] args) {
        Scanner scan = new Scanner(System.in);

        int n = scan.nextInt();
        int[] id = new int[n];

        for (int i = 0; i < n; i++) {
            id[i] = scan.nextInt();
        }

        Arrays.sort(id);

        int ans = 0, val = id[0], streak = 1;

        for (int i = 1; i < n; i++) {
            if (id[i] == val) {
                streak++;
            } else {
                if (val > 0 && streak > 2) {
                    System.out.println(-1);
                    return;
                }

                if (val > 0 && streak == 2) {
                    ans++;
                }

                val = id[i];
                streak = 1;
            }
        }

        if (val > 0 && streak > 2) {
            System.out.println(-1);
        } else {
            if (val > 0 && streak == 2) {
                //System.out.println(val);
                ans++;
            }

            System.out.println(ans);
        }
    }
}

File: 291/a.cpp
#include <cstdio>
#include <algorithm>

const int N = 1000;
int ids[N];

inline bool check_ok(int val, int streak) {
  return val > 0 && streak == 2;
}

inline bool check_fail(int val, int streak) {
  return val > 0 && streak > 2;
}

int main() {
  int n;
  scanf("%d", &n);

  for (int i = 0; i < n; ++i) {
    scanf("%d", ids + i);
  }

  std::sort(ids, ids + n);

  int ans = 0, val = ids[0], streak = 1;

  for (int i = 1; i < n; ++i) {
    if (ids[i] == val) {
      ++streak;
      continue;
    }

    if (check_fail(val, streak)) {
      printf("-1\n");
      return 0;
    }

    if (check_ok(val, streak)) {
      ++ans;
    }

    val = ids[i], streak = 1;
  }

  if (check_fail(val, streak)) {
    printf("-1\n");
  } else {
    if (check_ok(val, streak)) {
      ++ans;
    }

    printf("%d\n", ans);
  }

  return 0;
}
